src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -140,101 +123,66 @@ EFeatureType RandomFourierGaussPreproc::get_feature_type() {
 }
 
 int32_t RandomFourierGaussPreproc::get_dim_feature_space() const {
-	return ((int32_t) dim_feature_space);
+	return ((int32_t) m_dim_feature_space);
 }
 
 void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
 	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
+		error("InputError: feature space dimension must be greater 0!");
 	}
 
-	dim_feature_space = dim;
+	is_updated = m_dim_feature_space == dim;
+	SG_DEBUG("set_m_dim_feature_space(), is_updated = {}",is_updated);
+	m_dim_feature_space = dim;
 
 }
 
 int32_t RandomFourierGaussPreproc::get_dim_input_space() const {
-	return ((int32_t) dim_input_space);
+	return m_randomweights.num_cols;
 }
 
-void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
-	if (kernelwidth2 <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
+void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth ) {
+	if (kernelwidth <= 0) {
+		error("InputError: kernelwidth must be greater 0!");
 	}
-	kernelwidth=kernelwidth2;
-}
 
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
+	is_updated = m_kernelwidth == kernelwidth;
+	SG_DEBUG("set_kernelwidth(), is_updated = {}",is_updated);
+	m_kernelwidth=kernelwidth;
 }
 
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
-}
-
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
-
-			// already inited
-			return true;
-		} else {
-			return false;
-		}
-	}
-
-	return false;
+float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
+	return (m_kernelwidth);
 }
 
-bool RandomFourierGaussPreproc::init_randomcoefficients() {
-	if (dim_feature_space <= 0) {
-		throw ShogunException(
-				"bool CRandomFourierGaussPreproc::init_randomcoefficients(): dim_feature_space<=0 is not allowed\n");
+bool RandomFourierGaussPreproc::init_randomcoefficients(int32_t dim_input_space) {
+	if (m_dim_feature_space <= 0) {
+		error("ValueError: m_dim_feature_space has value <=0 which isn't allowed!, please set valid value via corresponding setter\n");
```

Comment:
these type of errors re not user facing, so no need for a message. I think since this is a helper method, no check is even needed. The caller has to make sure things are good. Put that in the docstring that it needs to be positive

Reply:
or assert